test_scripts/Defects/5_0/2379_SDL_must_send_log_error_and_ignore_invalid_RPC_in_case_SDL_cuts_off_fake_parameters_and_RPC_becomes_invalid.lua

```diff
@@ -0,0 +1,78 @@
+---------------------------------------------------------------------------------------------------
+-- User story: https://github.com/SmartDeviceLink/sdl_core/issues/2379
+--
+-- Precondition:
+-- SDL must send log error and ignore invalid RPC in case SDL cuts off fake parameters and RPC becomes invalid
+-- Description:
+-- Steps to reproduce:
+-- 1) SDL cuts off fake parameters from response (request) that SDL should use internally and this response (request) is invalid by any reason
+-- Expected:
+-- 1) Log the corresponding error
+-- 2) Ignore this response (request)
+---------------------------------------------------------------------------------------------------
+-- [[ Required Shared libraries ]]
+local runner = require('user_modules/script_runner')
+local common = require('user_modules/sequences/actions')
+
+-- [[ Test Configuration ]]
+runner.testSettings.isSelfIncluded = false
+
+-- [[ Local variables ]]
+local requestParamsValidData = {
+	menuID = 10,
+	position = 5,
+	menuName ="SubMenupositive",
+	fakeParam = 0001
+}
+local responseUiParamsValid = {
+	menuID = requestParamsValidData.menuID,
+	menuParams = {
+		position = requestParamsValidData.position,
+		menuName = requestParamsValidData.menuName		
+	}
+}
+local allParamsValid = {
+	requestParamsValidData = requestParamsValidData,
+	responseUiParamsValid = responseUiParamsValid
+}
+
+local requestParamsInvalidData = {
+	menuID = "10",
+	position = "5",
+	menuName = 123,
+	fakeParam = 0001
+}
+
+--[[ Local Functions ]]
+local function addSubMenuValid(pParams)
+	local cid = common.getMobileSession():SendRPC("AddSubMenu", pParams.requestParamsValidData)
+	common.getHMIConnection():ExpectRequest("UI.AddSubMenu", pParams.responseUiParamsValid)
+	:Do(function(_, data)
+		common.getHMIConnection():SendResponse(data.id, data.method, "SUCCESS", {})
+	end)
+	common.getMobileSession():ExpectResponse(cid, { success = true, resultCode = "SUCCESS" })
+	common.getMobileSession():ExpectNotification("OnHashChange")
+end 
+
+local function addSubMenuInvalidData(pParams)
+	local cid = common.getMobileSession():SendRPC("AddSubMenu", pParams.requestParamsInvalidData)
+	common.getHMIConnection():ExpectRequest("AddSubMenu")
+	:Times(0)
+	common.getMobileSession():ExpectResponse(cid, { success = false, resultCode = "INVALID_DATA" })	
+end
+
+--[[ Scenario ]]
+runner.Title("Preconditions")
+runner.Step("Clean environment", common.preconditions)
+runner.Step("Start SDL, HMI, connect Mobile, start Session", common.start)
+runner.Step("Register App", common.registerApp)
+runner.Step("Activate App", common.activateApp)
+
+-- [[ Test ]]
+runner.Title("Test")
+runner.Step("Add Submenu", addSubMenuValid, { allParamsValid })
+runner.Step("Add Submenu with Invalid Data", addSubMenuInvalidData ,{ requestParamsInvalidData })
```

Comment:
@TMyza Please fix logical error here: `requestParamsInvalidData` table does not contains field with name `requestParamsInvalidData` therefore code in line 58 is interpreted as 
```
local cid = common.getMobileSession():SendRPC("AddSubMenu", pParams.requestParamsInvalidData)
=> 
local cid = common.getMobileSession():SendRPC("AddSubMenu", nil)
```